File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.py
class Solution:
    def reverseWords(self, s: str) -> str:
        s = s.split()
        res = []
        for i in range(len(s)):
            w = swap(s[i])
            res.append("".join(w))
        return " ".join(res)


def swap(w):
    # word = [w[i] for i in range(len(w))]
    # Short version
    word = list(w)
    l, r = 0, len(word) - 1
    while l < r:
        temp = word[l]
        word[l] = word[r]
        word[r] = temp
        l += 1
        r -= 1
    return "".join(word)

File: leetcode_solutions/0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    s = s.split(" ");
    let res = []
    for(let i=0; i<s.length; i++){
        let word = swap(s[i])
        res.push(word);
     }
     return res.join(' ');

};

function swap(word){
  // let w = Array.from(word);
  // This also works
  let w = [...word];
  let l = 0;
  let r = w.length - 1;
  while (l < r){
    let temp = w[l];
    w[l] = w[r]
    w[r] = temp
    l += 1
    r -= 1
  }
  return w.join('')
};